### PR TITLE
update factorial

### PR DESCRIPTION
**Pull request by shellhub**
* add doctest
* less code
* support `0! = 1`

### CODE REVIEW

**Comment by cclauss** [comment]
Please add doctest for `factorial(-1)` which should raise `ValueError`.

https://docs.python.org/3/library/math.html#math.factorial
```
% python3
>>> import math
>>> math.factorial(-1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: factorial() not defined for negative values
```

**Comment by cclauss** [comment]
```
>>> math.factorial(0.1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: factorial() only accepts integral values
```

**Comment by shellhub** [comment]
@cclauss Please review again!

File: maths/factorial_python.py [hunk 1]
```diff
@@ -1,17 +1,27 @@
...
+    >>> factorial(0.1)
+    Traceback (most recent call last):
+        ...
+    ValueError: factorial() only accepts integral values
     """
 
     if input_number < 0:
-        raise ValueError("Input input_number should be non-negative")
+        raise ValueError("factorial() not defined for negative values")
+    if type(input_number) != int:
```
Comment (cclauss) [inline maths/factorial_python.py]: `if not isinstance(input_number, int):`

File: maths/factorial_recursive.py [hunk 2]
```diff
@@ -1,23 +1,30 @@
...
+    ValueError: factorial() not defined for negative values
+    >>> factorial(0.1)
+    Traceback (most recent call last):
+        ...
+    ValueError: factorial() only accepts integral values
     """
     if n < 0:
-        raise ValueError(n, "is negative")
+        raise ValueError("factorial() not defined for negative values")
+    if type(n) != int:
```
Comment (cclauss) [inline maths/factorial_recursive.py]: isinstance()

**Comment by shellhub** [comment]
@cclauss Please review again!

**Pull request merged**